server/src/internalClusterTest/java/org/elasticsearch/get/ShardMultiGetFomTranslogActionIT.java

```diff
@@ -93,19 +92,19 @@ public void testShardMultiGetFromTranslog() throws Exception {
         }
         assertThat(response.segmentGeneration(), equalTo(-1L));
         // Get followed by a Delete should still return a result
-        client().prepareDelete("test", "1").get();
-        response = getFromTranslog(indexOrAlias(), List.of("1", "2"));
+        var idToDelete = randomFrom(idsToIndex);
+        client().prepareDelete("test", idToDelete).get();
+        response = getFromTranslog(indexOrAlias(), idsToIndex);
         multiGetShardResponse = response.multiGetShardResponse();
         assertThat(getLocations(multiGetShardResponse).size(), equalTo(2));
         assertTrue(getFailures(multiGetShardResponse).stream().allMatch(Objects::isNull));
         getResponses = getResponses(multiGetShardResponse);
         assertThat(getResponses.size(), equalTo(2));
-        assertNotNull("get followed by a delete should still return a result", getResponses.get(0));
-        assertThat(getResponses.get(0).getId(), equalTo("1"));
-        assertThat(getResponses.get(0).isExists(), equalTo(false));
-        assertNotNull(getResponses.get(1));
-        assertThat(getResponses.get(1).getId(), equalTo("2"));
-        assertThat(getResponses.get(1).isExists(), equalTo(true));
+        assertTrue(getResponses.stream().allMatch(Objects::nonNull));
+        for (var getResponse : getResponses) {
```

Comment:
nit would be simpler to delete first element of idsToIndex. that way you'd be able to write directly assertions on the first and second element.

Reply:
The randomized simplified version is ok I think.